tests/test_arbitrage.py

```diff
@@ -16,20 +16,33 @@ def setUp(self):
 
     def test_check(self):
         q = arbitrage.Query(stock_names=self.names, api=self.api)
-        self.assertTrue(q.check(self.dates, self.option_prices, self.spot))
+        self.assertNotEqual(q.check(self.dates, self.option_prices, self.spot), [])
 
     def test_check_false(self):
         new_options = {"01/01/1990": {"call": 15, "put": 14, "strike": 9},
                        "01/02/1990": {"call": 15, "put": 15, "strike": 10}}
         new_data = arbitrage.StockData("VXX", self.dates, self.spot, new_options)
         new_api = arbitrage.FakeAPI(new_data)
         q = arbitrage.Query(stock_names=["VXX"], api=new_api)
-        self.assertFalse(q.check(self.dates, new_options, self.spot))
+        self.assertEqual(q.check(self.dates, new_options, self.spot), [])
+
+    def test_check_all_scenarios(self):
+        spot1, spot2 = 9, 10
+        below_strike_big_put = {"01/02/1990": {"call": 4, "put": 5, "strike": 10}}
+        below_strike_big_call = {"01/02/1990": {"call": 5, "put": 4, "strike": 10}}
+        above_strike_big_put = {"01/02/1990": {"call": 4, "put": 5, "strike": 9}}
+        above_strike_big_call = {"01/02/1990": {"call": 5, "put": 4, "strike": 9}}
+        new_dates = ["01/02/1990"]
+        q = arbitrage.Query(stock_names=["VXX"])
+        self.assertEqual(q.check(new_dates, below_strike_big_put, spot1), [])
+        self.assertNotEqual(q.check(new_dates, below_strike_big_call, spot1), [])
+        self.assertNotEqual(q.check(new_dates, above_strike_big_put, spot2), [])
+        self.assertEqual(q.check(new_dates, above_strike_big_call, spot2), [])
```

Comment:
This test is very hard to read. Consider refactoring.

At any rate, assertNotEqual is a bad assertion. Please provide what it should be equal to. 

What I want is a positive test for each scenario so you know its right for each (e.g. below strike & larger put, below strike & larger call, etc). 

Maybe a negative test for each scenario too.

Reply:
fixed tests